GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -1230,20 +1239,20 @@ private void OnGridViewSelectionChanged(object sender, EventArgs e)
             _selectionTimer.Enabled = true;
             _selectionTimer.Start();
 
-            var selectedRevisions = GetSelectedRevisions();
-            var firstSelectedRevision = selectedRevisions.FirstOrDefault();
-            var secondSelectedRevision = selectedRevisions.Skip(1).FirstOrDefault();
-
-            if (selectedRevisions.Count == 1 && firstSelectedRevision != null)
-            {
-                _navigationHistory.Push(firstSelectedRevision.ObjectId);
-            }
+            var (first, selected) = getFirstAndSelected();
 
-            compareToWorkingDirectoryMenuItem.Enabled = firstSelectedRevision != null && firstSelectedRevision.ObjectId != ObjectId.WorkTreeId;
+            compareToWorkingDirectoryMenuItem.Enabled = selected != null && selected.ObjectId != ObjectId.WorkTreeId;
             compareWithCurrentBranchToolStripMenuItem.Enabled = Module.GetSelectedBranch(setDefaultIfEmpty: false).IsNotNullOrWhitespace();
-            compareSelectedCommitsMenuItem.Enabled = firstSelectedRevision != null && secondSelectedRevision != null;
+            compareSelectedCommitsMenuItem.Enabled = first != null && selected != null;
+            openCommitsWithDiffToolMenuItem.Enabled = first != null && selected != null; //// && !first.IsArtificial && !selected.IsArtificial;
```

Comment:
Shall the dead code be kept?

Reply:
I added that just to let you find something more!
Remains from WinMerge compatibility - it will be hard to explain this.